metrics/cmd/heapster-apiserver/app/server.go

```diff
@@ -85,78 +60,35 @@ func NewHeapsterApiServer(s *options.HeapsterRunOptions, metricSink *metricsink.
 	}, nil
 }
 
-func newAPIServer(s *genericoptions.ServerRunOptions) (*genericapiserver.GenericAPIServer, error) {
-	genericapiserver.DefaultAndValidateRunOptions(s)
-
-	resourceConfig := genericapiserver.NewResourceConfig()
-
-	storageGroupsToEncodingVersion, err := s.StorageGroupsToEncodingVersion()
-	if err != nil {
-		glog.Fatalf("error generating storage version map: %s", err)
-	}
-	storageFactory, err := genericapiserver.BuildDefaultStorageFactory(
-		s.StorageConfig, s.DefaultStorageMediaType, api.Codecs,
-		genericapiserver.NewDefaultResourceEncodingConfig(), storageGroupsToEncodingVersion,
-		[]unversioned.GroupVersionResource{}, resourceConfig, s.RuntimeConfig)
-	if err != nil {
-		glog.Fatalf("error in initializing storage factory: %s", err)
+func newAPIServer(s *options.HeapsterRunOptions) (*genericapiserver.GenericAPIServer, error) {
```

Comment:
done, the rewiring stuff having to do with the generic API server code is now in it's own commit.